tests/conftest.py

```diff
@@ -0,0 +1,2487 @@
+import contextlib
+import os
+import sys
+from datetime import datetime
+from unittest.mock import PropertyMock, patch
+
+import numpy as np
+import pandas as pd
+import py
+import pytest
+import woodwork as ww
+from sklearn import datasets
+from sklearn.preprocessing import minmax_scale
+from skopt.space import Integer, Real
+from woodwork import logical_types as ww_logical_types
+from woodwork.logical_types import (
+    AgeNullable,
+    BooleanNullable,
+    IntegerNullable,
+    Ordinal,
+)
+
+from evalml.demos import load_weather
+from evalml.model_family import ModelFamily
+from evalml.objectives import BinaryClassificationObjective
+from evalml.objectives.utils import get_core_objectives, get_non_core_objectives
+from evalml.pipelines import (
+    BinaryClassificationPipeline,
+    MulticlassClassificationPipeline,
+    RegressionPipeline,
+    TimeSeriesBinaryClassificationPipeline,
+    TimeSeriesMulticlassClassificationPipeline,
+    TimeSeriesRegressionPipeline,
+)
+from evalml.pipelines.components import (
+    DecisionTreeClassifier,
+    DecisionTreeRegressor,
+    Estimator,
+    LogisticRegressionClassifier,
+)
+from evalml.pipelines.components.component_base import ComponentBase
+from evalml.pipelines.components.ensemble.stacked_ensemble_base import (
+    _nonstackable_model_families,
+)
+from evalml.pipelines.components.utils import _all_estimators
+from evalml.preprocessing import load_data
+from evalml.problem_types import (
+    ProblemTypes,
+    handle_problem_types,
+    is_binary,
+    is_multiclass,
+    is_regression,
+    is_time_series,
+)
+from evalml.utils import infer_feature_types
+from evalml.utils.cli_utils import get_evalml_black_config
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "skip_offline: mark test to be skipped if offline (https://oss.alteryx.com cannot be reached)",
+    )
+    config.addinivalue_line(
+        "markers",
+        "skip_during_conda: mark test to be skipped if running during conda build",
+    )
+
+
+@pytest.fixture(scope="session")
+def go():
+    from plotly import graph_objects as go
+
+    return go
+
+
+@pytest.fixture(scope="session")
+def im():
+    from imblearn import over_sampling as im
+
+    return im
+
+
+@pytest.fixture(scope="session")
+def lgbm():
```

Comment:
There's a lot of functions in the conftest file that we don't use at all. I would start cleaning this up and removing functions unused, since once we have codecov in, it'll raise errors